Source/ExcelDna.Integration/ExcelComAddIn.cs

```diff
@@ -47,6 +47,7 @@ public virtual void OnConnection(object Application, ext_ConnectMode ConnectMode
 
         public virtual void OnDisconnection(ext_DisconnectMode RemoveMode, ref Array custom)
         {
+            SynchronizationManager.Uninstall();
```

Comment:
Should this rather be called in OnBeginShutdown? We might have different ComAddIn objects loaded, and they might be unloaded without Excel closing while we want the rest of the add-in to keep working. 
We're basically trying to use whatever ExcelComAddIn object is created (for a ribbon, as a CTP helper or just by the user) and use that to get an Excel Shutdown signal. The user might override the event ExcelComAddIn handler in their code and not call the base method, or there might be multiple such objects running in a session.

Reply:
Maybe we can also add a comment explaining this in the code.

Reply:
Can you give an example of an ComAddIn object unloaded without Excel closing, so I could experiment?